docs/src/main/paradox/user/lang/kotlin/eventsourced.md

```diff
@@ -0,0 +1,103 @@
+
+@@snip [ShoppingCartEntity.kt](/docs/src/test/kotlin/docs/user/eventsourced/ShoppingCartEntity.kt) { #entity-state }
+
+## Constructing
+
+The context available for injection into the constructor is a `EventSourcedEntityCreationContext` (io.cloudstate.javasupport.eventsourced.EventSourcedEntityCreationContext). While you don't necessarily need to define a constructor, you can define one and have that context injected in. The constructor below shows having the entity id injected:
+
+@@snip [ShoppingCartEntity.kt](/docs/src/test/kotlin/docs/user/eventsourced/ShoppingCartEntity.kt) { #constructing }
+
+## Handling commands
+
+Command handlers can be declared by annotating a method with `@CommandHandler` (io.cloudstate.kotlinsupport.api.eventsourced.CommandHandler). They take a context class of type `CommandContext` (io.cloudstate.javasupport.eventsourced.CommandContext).
+
+By default, the name of the command that the method handles will be the name of the method with the first letter capitalized. So, a method called `getCart` will handle gRPC service call command named `GetCart`. This can be overridden by setting the `name` parameter on the `@CommandHandler` annotation.
+
+The command handler also can take the gRPC service call input type as a parameter, to receive the command message. This is optional, sometimes it's not needed, for example, our `GetCart` service call doesn't need any information from the message, since it's just returning the current state as is. Meanwhile, the `AddItem` service call does need information from the message, since it needs to know the product id, description and quantity to add to the cart.
+
+The return type of the command handler must be the output type for the gRPC service call, this will be sent as the reply.
+
+The following shows the implementation of the `GetCart` command handler. This command handler is a read-only command handler, it doesn't emit any events, it just returns some state:
+
+@@snip [ShoppingCartEntity.kt](/docs/src/test/kotlin/docs/user/eventsourced/ShoppingCartEntity.kt) { #get-cart }
+
+### Emitting events
+
+Commands that modify the state may do so by emitting events.
+
+@@@ warning
+The **only** way a command handler may modify its state is by emitting an event. Any modifications made directly to the state from the command handler will not be persisted, and when the entity is passivated and next reloaded, those modifications will not be present.
+@@@
+
+A command handler may emit an event by taking in a `CommandContext` (io.cloudstate.javasupport.eventsourced.CommandContext) parameter, and invoking the `emit` method on it. Invoking `emit` will immediately invoke the associated event handler for that event - this both validates that the event can be applied to the current state, as well as updates the state so that subsequent processing in the command handler can use it.
```

Comment:
This is how the CommandContext looks in the browser; similar in other places. Perhaps you could reformat or even omit the package?

<img width="892" alt="Screenshot 2020-03-05 at 00 09 55" src="https://user-images.githubusercontent.com/12616445/75931596-aaec0300-5e75-11ea-8a72-eeea3cab1df3.png">

Reply:
Ok Marcel can omit the package scheme. But that was really the intention here.

Reply:
@sleipnir please leave it if you like. My comment was purely because of it looks not that fluent in text. You could perhaps try to have variations?

Reply:
No problem @marcellanz the java documentation takes you to the javadoc and becomes fluent. I'm not sure if I want to do this right now, I should probably just mark it in bold